### PR TITLE
use `pkgconf` instead of `pkg-config` when on bsd in imagemagick package

### PR DESCRIPTION
**Pull request by Phate6660**

### CODE REVIEW

File: packages/i/imagemagick/xmake.lua [hunk 1]
```diff
@@ -46,9 +46,19 @@ package("imagemagick")
...
             package:add("syslinks", "pthread")
         end
+
+        if package:is_plat("linux") then
+            package:add("deps", "pkg-config")
+        end
+
+        if package:is_plat("bsd") then
+            package:add("deps", "pkgconf")
+            package:set_env("PKG_CONFIG", path.join(packackage:dep("pkgconf"):installdir("bin"), "pkgconf"))
```
Comment (waruqi) [inline packages/i/imagemagick/xmake.lua:60]: package:addenv

File: packages/i/imagemagick/xmake.lua [hunk 2]
```diff
@@ -46,9 +46,19 @@ package("imagemagick")
...
             package:add("syslinks", "pthread")
         end
+
+        if package:is_plat("linux") then
+            package:add("deps", "pkg-config")
+        end
+
+        if package:is_plat("bsd") then
+            package:add("deps", "pkgconf")
+            package:add_env("PKG_CONFIG", path.join(packackage:dep("pkgconf"):installdir("bin"), "pkgconf"))
```
Comment (waruqi) [inline packages/i/imagemagick/xmake.lua:60]: not add_env,  is addenv

Comment (waruqi) [reply packages/i/imagemagick/xmake.lua:60]:
```lua
local pkgconf = package:find_tool("pkgconf")
if pkgconf then
    package:addenv("PKG_CONFIG", pkgconf.program)
end
```

**Comment by Phate6660** [comment]
That Fedora CI was just due to a timeout when downloading the tarball.
Given that all the others passed, it's probably safe to say that would have too.
But we could run the CI again.

Besides that, it seems like the BSD CIs passed!

**Pull request merged**